src/main/java/picard/fingerprint/CheckFingerprint.java

```diff
@@ -229,28 +229,34 @@ protected int doWork() {
             outputSummaryMetricsFile = new File(OUTPUT + FINGERPRINT_SUMMARY_FILE_SUFFIX);
         }
 
-        IOUtil.assertFileIsReadable(INPUT);
+        try {
+            inputPath = IOUtil.getPath(INPUT);
+            genotypesPath = IOUtil.getPath(GENOTYPES);
+        } catch (IOException e) {
+            throw new IllegalArgumentException(e);
+        }
```

Comment:
seems like a hack so you don't have to change `IOUtil.assertFileIsReadable`.  Is this going to become the new normal for all of Picard if we want GCS support (it's a slippery slope).

Reply:
My hope is that we can come up with a wrapper of sorts that behaves like a Path but has a string constructor (which Path doesn't have) so it can be used as an `@Argument`. Until that time, we will need to use this "hack"...I think.

Reply:
It should be pretty easy to add support for `Path` in Barclay similar to how it has support for `Enum`: https://github.com/broadinstitute/barclay/blob/master/src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java#L1108

Even more interesting is that Barclay could handle `Optional` rather than relying on `null`, but that's for another day.

We do this for fgbio via our [commons](https://github.com/fulcrumgenomics/commons/blob/master/src/main/scala/com/fulcrumgenomics/commons/reflect/ReflectionUtil.scala#L370-L408) library.  Barclay isn't much different in how it goes about constructing arguments from strings.